### PR TITLE
Reduce number of SocketServer threads

### PR DESCRIPTION
**Pull request by snalli**
We don't use socket server code but we still end up creating 100s of thread just sitting idle.
This pollutes thread dumps. So let's just reduce the number of socket server threads to 1.

### CODE REVIEW

File: ambry-server/src/main/java/com/github/ambry/server/AmbryServer.java [hunk 1]
```diff
@@ -308,7 +308,7 @@ public void startup() throws InstantiationException {
       requests = new AmbryServerRequests(storageManager, networkServer.getRequestResponseChannel(), clusterMap, nodeId,
           registry, metrics, findTokenHelper, notificationSystem, replicationManager, storeKeyFactory, serverConfig,
           diskManagerConfig, storeKeyConverterFactory, statsManager, clusterParticipants.get(0));
-      requestHandlerPool = new RequestHandlerPool(serverConfig.serverRequestHandlerNumOfThreads,
+      requestHandlerPool = new RequestHandlerPool(1,
```
Comment (justinlin-linkedin) [inline ambry-server/src/main/java/com/github/ambry/server/AmbryServer.java:311]: let's do it through configuration, instead of hard coding.

Comment (snalli) [reply ambry-server/src/main/java/com/github/ambry/server/AmbryServer.java]: done

File: ambry-server/src/main/java/com/github/ambry/server/AmbryServer.java [hunk 2]
```diff
@@ -308,7 +308,7 @@ public void startup() throws InstantiationException {
       requests = new AmbryServerRequests(storageManager, networkServer.getRequestResponseChannel(), clusterMap, nodeId,
           registry, metrics, findTokenHelper, notificationSystem, replicationManager, storeKeyFactory, serverConfig,
           diskManagerConfig, storeKeyConverterFactory, statsManager, clusterParticipants.get(0));
-      requestHandlerPool = new RequestHandlerPool(serverConfig.serverRequestHandlerNumOfThreads,
+      requestHandlerPool = new RequestHandlerPool(serverConfig.serverRequestHandlerNumSocketServerThreads,
```
Comment (Arun-LinkedIn) [inline ambry-server/src/main/java/com/github/ambry/server/AmbryServer.java:311]: nit: We could also use this config in VCR Server too - https://github.com/linkedin/ambry/blob/9c685c5389ff6c07523a0cba89c16781b882a4b7/ambry-cloud/src/main/java/com/github/ambry/cloud/VcrServer.java#L245

**Review by Arun-LinkedIn — APPROVED** [review]

**Review by justinlin-linkedin — APPROVED** [review]

File: ambry-server/src/main/java/com/github/ambry/server/AmbryServer.java (hunk 2 above)
Comment (snalli) [reply ambry-server/src/main/java/com/github/ambry/server/AmbryServer.java:311]: done

**Pull request merged**